doc/api/worker.md

```diff
@@ -0,0 +1,343 @@
+`require('worker')` itself.
+
+## worker.isMainThread
+<!-- YAML
+added: REPLACEME
+-->
+
+* {boolean}
+
+Is `true` if this code is not running inside of a [`Worker`][] thread.
+
+## worker.postMessage(value[, transferList])
+<!-- YAML
+added: REPLACEME
+-->
+
+This method is available if this is a `Worker` thread, which can be tested
+using [`require('worker').isMainThread`][].
+
+Send a message to the parent thread’s `Worker` instance that will be received
+via [`worker.on('message')`][]. See [`port.postMessage()`][] for
+more details.
+
+## worker.threadId
+<!-- YAML
+added: REPLACEME
+-->
+
+* {integer}
+
+An integer identifier for the current thread. On the corresponding worker object
+(if there is any), it is available as [`worker.threadId`][].
+
+## worker.workerData
+<!-- YAML
+added: REPLACEME
+-->
+
+An arbitrary JavaScript object that contains a clone of the data passed
+to this thread’s `Worker` constructor.
+
+
+## Class: MessageChannel
+<!-- YAML
+added: REPLACEME
+-->
+
+The `MessageChannel` has no methods of its own. `new MessageChannel()`
+yields an object with `port1` and `port2` properties, which refer to linked
+[`MessagePort`][] instances.
+
+```js
+const { MessageChannel } = require('worker');
+
+const { port1, port2 } = new MessageChannel();
+port1.on('message', (message) => console.log('received', message));
+port2.postMessage({ foo: 'bar' });
+// prints: received { foo: 'bar' }
+```
+
+## Class: MessagePort
+<!-- YAML
+added: REPLACEME
+-->
+
+* Extends: {EventEmitter}
+
+Instances of the `worker.MessagePort` class represent an asynchronous
+communications channel. It can be used to transfer structured data,
+memory regions and other `MessagePort`s between different [`Worker`][]s
+or [`vm` context][vm]s.
+
+For transferring `MessagePort` instances between VM contexts, see
+[`vm.moveMessagePortToContext()`][].
+
+*Note*: With the exception of `MessagePort`s being [`EventEmitter`][]s rather
+than `EventTarget`s, this implementation matches [browser `MessagePort`][]s.
+
+### Event: 'close'
+<!-- YAML
+added: REPLACEME
+-->
+
+The `'close'` event is emitted once either side of the channel has been
+disconnected.
+
+### Event: 'error'
+<!-- YAML
+added: REPLACEME
+-->
+
+The `'error'` event is emitted if the worker thread throws an uncaught
+expection. In that case, the worker will be terminated.
+
+### Event: 'message'
+<!-- YAML
+added: REPLACEME
+-->
+
+The `'message'` event is emitted for any incoming message, containing the cloned
+input of [`port.postMessage()`][].
+
+### port.close()
+<!-- YAML
+added: REPLACEME
+-->
+
+* Returns: {undefined}
+
+Disables further sending of messages on either side of the connection.
+
+### port.postMessage(value[, transferList])
+<!-- YAML
+added: REPLACEME
+-->
+
+* Returns: {undefined}
+
+Sends a JavaScript value to the receiving side of this channel.
+`value` will be transferred in a way
+that is compatible with the [HTML structured clone algorithm][]. In particular,
```

Comment:
structured cloning isn't a thing any more (https://html.spec.whatwg.org/multipage/structured-data.html#structuredclone).

Reply:
Heh. Right. I think this is still what it’s known as, though? (That it’s implemented as a serialize + deserialize step should be more or less transparent to users)

Reply:
@addaleax That article is pretty outdated though (it has no mention of MessagePort or SharedArrayBuffer)... HTML structured serialization/deserialization? maybe a link to v8.Serializer/Deserializer?

Reply:
> (it has no mention of MessagePort or SharedArrayBuffer)

That’s why these are mentioned explicitly here ;)

> maybe a link to v8.Serializer/Deserializer?

Yeah, that’s a good idea … that should give people an idea of how to play around with the algorithm without spinning up message channels every time. I’ve added:

```diff
+For more information on the serialization and deserialization mechanisms
+behind this API, see the [serialization API of the `v8` module][v8.serdes].
```